### PR TITLE
Another way of trying to get sonar to work

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Bugfix

### Description

Fixes active sonar scanning 

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -53,4 +53,4 @@ jobs:
       - name: 'Build example service'
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        run: mvn -B clean verify -Pci
+        run: mvn -B clean verify -Pci -DDsonar.projectKey=kroxylicious_kroxylicious
```
Comment (k-wall) [inline .github/workflows/maven.yml:56]: Double D looks fishy!

Comment (SamBarker) [reply .github/workflows/maven.yml:56]: Doh! Good spotting

File: pom.xml [hunk 2]
```diff
@@ -458,6 +459,16 @@
                     <artifactId>build-helper-maven-plugin</artifactId>
                     <version>3.4.0</version>
                 </plugin>
+                <plugin>
+                    <groupId>org.sonarsource.scanner.maven</groupId>
```
Comment (k-wall) [inline pom.xml:463]: Do you think sonar should be in a separate profile?  I'm thinking about people who might wish to build Kroxy themselves in a CI system (derivates builds), but wouldn't necessarily want to hook up to Sonar.

**Review by k-wall — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (SamBarker) [reply pom.xml:463]: I don't think the CI profile represents anything other than our CI usage so no I don't. I'll happily shuffle things around if and when someone asks us too.

**Review by robobario — APPROVED** [review]
LGTM

**Pull request merged**